cmd/app.go

```diff
@@ -238,30 +260,90 @@ func appLs(ctx *cli.Context) error {
 	}
 
 	collection, err := c.ProjectClient.App.List(defaultListOpts(ctx))
+	if err != nil {
+		return err
+	}
+
 	writer := NewTableWriter([][]string{
 		{"ID", "ID"},
 		{"NAME", "App.Name"},
 		{"STATE", "App.State"},
+		{"CATALOG", "Catalog"},
+		{"TEMPLATE", "Template"},
 		{"VERSION", "Version"},
 	}, ctx)
 
 	defer writer.Close()
 
 	for _, item := range collection.Data {
-		parsedExternal, err := parseExternalID(item.ExternalID)
+		appData := &AppData{
+			ID:  item.ID,
+			App: item,
+		}
+
+		appExternalID := item.ExternalID
+		var appTemplateFiles map[string]string
+		if appExternalID == "" {
+			appRevision, err := c.ProjectClient.AppRevision.ByID(strings.Replace(item.ID, item.Name, item.AppRevisionID, -1))
```

Comment:
I know this is for generating a namespaced revisionId whereas we don't have a namespace prefix in the appRevisionID field. But it looks confusing. Extract the `strings.Replace` to a single line and add a comment for it, or fetch the namespace from `projectID` would be more descriptive.

Reply:
Adjusted